config.ts

```diff
@@ -129,6 +145,57 @@ interface HomePage {
 const homePage = z.strictObject({
   sections: z.array(homeSection),
 });
+
+// Listing Types
+export enum ListingType {
+  Developer = 'Developer',
+  Creative = 'Creative',
+  Managerial = 'Managerial',
+  Volunteer = 'Volunteer',
+}
+
+const listingTypeEnum = z.nativeEnum(ListingType);
+
+interface Requirement {
+  description: string;
+  icon_path?: string;
+}
+
+const requirement = z.strictObject({
+  description: z.string(),
+  icon_path: z.optional(z.string()),
+});
+
+export interface Listing {
+  priority?: number;
+  title: string;
+  description: string;
+  requirements: Requirement[];
+  type: (typeof ListingType)[keyof typeof ListingType];
+  modal?: string;
+  keywords?: string[];
+}
+
+const listing = z.strictObject({
+  priority: z.number().optional(),
+  title: z.string(),
+  description: z.string(),
+  requirements: z.array(requirement),
+  type: listingTypeEnum,
+  modal: z.string().optional(),
+  keywords: z.array(z.string()).optional(),
```

Comment:
Makes sense and fixed it but just to give some clarification as to why I did it this way. Some non-devs will touch the config. Will they know what an array is?

Reply:
> Makes sense and fixed it but just to give some clarification as to why I did it this way. Some non-devs will touch the config. Will they know what an array is?

I think it is fair to make a basic assumption that they have a json level understanding of yaml coming in. by that I mean they understand the basic syntax and data layout but dont need to understand the `*text` and `&text` stuff and things like that. I think it's reasonable that we are available to explain the basics of using yaml or at least can provide some documentation. 

simplifying furthur then that is going to have problems because you need to know some level of code to write a config.